webclient/core/import_data_mappers.py

```diff
@@ -0,0 +1,1522 @@
+        return field_mapping
+
+
+class NeidentAkceVedouciMapper(ImportModelMapper):
+    fields = tuple()
+    model_class = NeidentAkceVedouci
+    primary_key = ("neident_akce", "vedouci")
+
+    @classmethod
+    def get_mapping(cls):
+        field_mapping = super().get_mapping()
+        field_mapping["neident_akce"] = LookupImportField(DokumentCast)
+        field_mapping["vedouci"] = LookupImportField(Osoba)
+        return field_mapping
+
+
+class KomponentaMapper(ImportModelMapper):
+    fields = ("ident_cely", "jistota", "presna_datace", "poznamka")
+    model_class = Komponenta
+    primary_key = "ident_cely"
+
+    @classmethod
+    def get_mapping(cls):
+        field_mapping = super().get_mapping()
+        field_mapping["vazba"] = KomponentaLookupImportField(
+            [DokumentacniJednotka, DokumentCast], read_field_name="komponenty"
+        )
+        field_mapping["obdobi"] = LookupImportField(Heslar)
+        field_mapping["areal"] = LookupImportField(Heslar)
+        return field_mapping
+
+
+class KomponentaAktivitaMapper(ImportModelMapper):
+    fields = ()
+    model_class = KomponentaAktivita
+    primary_key = ("komponenta", "aktivita")
+
+    @classmethod
+    def get_mapping(cls):
+        field_mapping = super().get_mapping()
+        field_mapping["komponenta"] = LookupImportField(Komponenta)
+        field_mapping["aktivita"] = LookupImportField(Heslar)
+        return field_mapping
+
+
+class NalezMapper(ImportModelMapper):
+    fields = ("id", "pocet", "poznamka")
+    primary_key = "id"
+
+    @classmethod
+    def get_mapping(cls):
+        field_mapping = super().get_mapping()
+        field_mapping["komponenta"] = LookupImportField(Komponenta)
+        field_mapping["druh"] = LookupImportField(Heslar)
+        field_mapping["specifikace"] = LookupImportField(Heslar)
+        return field_mapping
+
+
+class NalezObjektMapper(NalezMapper):
+    model_class = NalezObjekt
+
+
+class NalezPredmetMapper(NalezMapper):
+    model_class = NalezPredmet
+
+
+class ExterniZdrojMapper(ImportModelMapper):
+    fields = (
+        "ident_cely",
+        "doi",
+        "stav",
+        "rok_vydani_vzniku",
+        "nazev",
+        "sbornik_nazev",
+        "edice_rada",
+        "casopis_denik_nazev",
+        "casopis_rocnik",
+        "isbn",
+        "issn",
+        "misto",
+        "vydavatel",
+        "datum_rd",
+        "paginace_titulu",
+        "link",
+        "organizace",
+        "poznamka",
+    )
+    model_class = ExterniZdroj
+    primary_key = "ident_cely"
+
+    @classmethod
+    def get_mapping(cls):
+        field_mapping = super().get_mapping()
+        field_mapping["typ"] = LookupImportField(Heslar)
+        field_mapping["typ_dokumentu"] = LookupImportField(Heslar)
+        return field_mapping
+
+
+class ExterniZdrojOsobaMapper(ImportModelMapper):
+    fields = ("poradi",)
+    primary_key = ("externi_zdroj", "autor")
```

Comment:
Toto ale neplatí pro `externi_zdroje_editori` (je tam pole `editor`, ne `autor`). Asi bude snazší napsat to jako dvě podobné třídy zvlášť (dotýká se i dalších míst níže).

Reply:
Rozepsáno do samostatných tříd.